### PR TITLE
[STK-376] Add test cases

### PR DESCRIPTION
**Pull request by faizanahmad055**
Added unit test cases for kube package
Added unit test cases for dockergen funcs

### CODE REVIEW

File: pkg/testutil/kube.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+							Name: statefulsetName,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getkonfiguratorTemplate() v1alpha1.KonfiguratorTemplate {
```
Comment (golangci) [inline pkg/testutil/kube.go]: `getkonfiguratorTemplate` is unused

File: pkg/template/funcs/dockergen_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+	if expected != result {
+		t.Errorf("MapContains should return true, since 'thirdKey' is present in the map, expected: '%v', actual: '%v'", expected, result)
+	}
+}
+
+func TestWhen(t *testing.T) {
+	firstValue := "first"
+	secondValue := "first"
+	if result := when(firstValue == secondValue, true, false); result.(bool) != true {
```
Comment (golangci) [inline pkg/template/funcs/dockergen_test.go]: should omit comparison to bool constant, can be simplified to !result.(bool)

File: pkg/template/funcs/dockergen_test.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+		t.Errorf("'When' should return true, since initial condition is true, expected: '%v', actual: '%v'", true, result)
+	}
+
+	secondValue = "second"
+	if result := when(firstValue == secondValue, true, false); result.(bool) != false {
```
Comment (golangci) [inline pkg/template/funcs/dockergen_test.go]: should omit comparison to bool constant, can be simplified to result.(bool)

File: pkg/template/funcs/dockergen_test.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+		if value != "first" && value != "second" {
+			t.Errorf("intersect did not return correct values, expected: 'first' or 'second, found: '%v'", value)
+		}
+	}
+}
+
+func TestExists(t *testing.T) {
+	path := "../../template/funcs/funcs.go"
+	result, _ := exists(path)
+	if result != true {
```
Comment (golangci) [inline pkg/template/funcs/dockergen_test.go]: should omit comparison to bool constant, can be simplified to !result

**Comment by stakater-user** [comment]
@faizanahmad055 Image is available for testing. ``docker pull stakater/konfigurator:PR-5-1``

**Comment by stakater-user** [comment]
@faizanahmad055 Image is available for testing. ``docker pull stakater/konfigurator:PR-5-2``

**Comment by stakater-user** [comment]
@faizanahmad055 Image is available for testing. ``docker pull stakater/konfigurator:PR-5-3``

**Comment by stakater-user** [comment]
@faizanahmad055 Image is available for testing. ``docker pull stakater/konfigurator:PR-5-4``

File: pkg/testutil/kube.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
+package testutil
```
[the remark this answers is not in the document]
Comment (faizanahmad055) [reply pkg/testutil/kube.go]: Updated

**Pull request merged**